### PR TITLE
Add scripts and modify CMakelists.txt to build on spock and crusher

### PR DESCRIPTION
**Pull request by mewall**
This change is [Reviewable](https://reviewable.io/reviews/lanl/qmd-progress/218)

### CODE REVIEW

**Review by cnegre — APPROVED** [review]

**Review by suemni — APPROVED** [review]

**Review by jeanlucf22 — CHANGES_REQUESTED** [review]
I personally don't like adding specific compiler flags in CMakeList.txt. Modern cmake takes care of that. If you need to change the default chosen by cmake (which I understand may happen), use the "EXTRA" flags. What flags would cmake pick for you on spock and crusher?

**Comment by mewall** [comment]
As I recall those edits were made to include the "-ef" flag for Fortran.


The CMakeLists.txt edits follow what's already done for Intel and GNU SDKs.


Might it make sense to merge and make this a separate issue to address, changing all three?


Along similar lines, it would be nice to implement the include(FindBLAS) method as was done for bml...

________________________________
From: Jean-Luc Fattebert ***@***.***>
Sent: Friday, February 18, 2022 5:37:07 AM
To: lanl/qmd-progress
Cc: Wall, Michael E; Author
Subject: [EXTERNAL] Re: [lanl/qmd-progress] Add scripts and modify CMakelists.txt to build on spock and crusher (PR #218)


@jeanlucf22 requested changes on this pull request.

I personally don't like adding specific compiler flags in CMakeList.txt. Modern cmake takes care of that. If you need to change the default chosen by cmake (which I understand may happen), use the "EXTRA" flags. What flags would cmake pick for you on spock and crusher?

**Comment by jeanlucf22** [comment]
@mewall  Why don't you use EXTRA_FCFLAGS to add "-ef"? IMO all the compiler flags in CMakeList.txt should disappear at some point, so let's not add another one...

BLAS: you should be picking up the one used to build BML (with pkg_check_modules(BML REQUIRED bml) ) and not need to set it up explicitly... That is why we don't have anything about BLAS inCMakeList.txt. Same for Lapack and Magma.

**Comment by mewall** [comment]
***@***.***<https://github.com/mewall> Why don't you use EXTRA_FCFLAGS to add "-ef"? IMO all the compiler flags in CMakeList.txt should disappear at some point, so let's not add another one...


Those edits were to enable builds using CCE (useful, e.g. for offload builds). Since this PR is mainly intended to address MAGMA builds, which use GNU SDK, I will restore the CMakeLists.txt file to what's in master.


BTW, in my last commit to BML there was a request to clean up the build script by removing machine-specific flags...this is why I had put the CCE mods in CMakeLists.txt...


>BLAS: you should be picking up the one used to build BML (with pkg_check_modules(BML REQUIRED bml) ) and not need to set it up explicitly... That is why we don't have anything about BLAS inCMakeList.txt. Same for Lapack and Magma.


You are right, those extra flags aren't needed (at least now, probably I put them in because they were part of some modifications in the past to make it work). I'll take them out.

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -28,7 +28,8 @@ endfunction(progress_example)
 progress_example(changecoords changecoords/changecoords.F90)
 progress_example(getdihedral getdihedral/getdihedral.F90)
 progress_example(gpscf_dist gpdist/gpscf_dist.F90)
-progress_example(gpmd_dist gpdist/gpmd_dist.F90)
+#MEW Comment out as doesn't build on crusher
```
Comment (jeanlucf22) [inline examples/CMakeLists.txt:31]: Do we want to turn off that one because it doesn't build on one platform? Why does it not build?

**Comment by mewall** [comment]
It's Christian's code. He said to go ahead and comment it out, since there's new example code being prepared to replace it.
________________________________
From: Jean-Luc Fattebert ***@***.***>
Sent: Friday, February 18, 2022 10:02:01 AM
To: lanl/qmd-progress
Cc: Wall, Michael E; Mention
Subject: [EXTERNAL] Re: [lanl/qmd-progress] Add scripts and modify CMakelists.txt to build on spock and crusher (PR #218)


@jeanlucf22 commented on this pull request.

________________________________

In examples/CMakeLists.txt<https://github.com/lanl/qmd-progress/pull/218#discussion_r810183699>:

> @@ -28,7 +28,8 @@ endfunction(progress_example)
 progress_example(changecoords changecoords/changecoords.F90)
 progress_example(getdihedral getdihedral/getdihedral.F90)
 progress_example(gpscf_dist gpdist/gpscf_dist.F90)
-progress_example(gpmd_dist gpdist/gpmd_dist.F90)
+#MEW Comment out as doesn't build on crusher


Do we want to turn off that one because it doesn't build on one platform? Why does it not build?

**Review by jeanlucf22 — APPROVED** [review]

**Review by nicolasbock — APPROVED** [review]

**Pull request merged**